src/main/java/sample/stream/GroupLogFile.java

```diff
@@ -32,37 +30,35 @@ public static void main(String[] args) throws IOException {
     final String inPath = "src/main/resources/logfile.txt";
     final File inputFile = new File(inPath);
 
-    SynchronousFileSource.create(inputFile).
-        // parse bytestrings (chunks of data) to lines
-        via(Framing.delimiter(ByteString.fromString(System.lineSeparator()), 512, true)).
-        map(bytestring -> bytestring.utf8String()).
-        // group them by log level
-        groupBy(line -> {
-          final Matcher matcher = loglevelPattern.matcher(line);
-          if (matcher.find())
-            return matcher.group(1);
-          else
-            return "OTHER";
-        }).
-        // write lines of each group to a separate file
-        runForeach(levelProducerPair -> {
-          final String outPath = "target/log-" + levelProducerPair.first() + ".txt";
-          final PrintWriter output = new PrintWriter(new FileOutputStream(outPath), true);
+    final Map<String, PrintWriter> outputs = new HashMap<>();
+    for (String level : Arrays.asList("DEBUG", "INFO", "WARN", "ERROR", "UNKNOWN")) {
+      final String outPath = "target/log-" + level + ".txt";
+      final PrintWriter output = new PrintWriter(new FileOutputStream(outPath), true);
+      outputs.put(level, output);
+    }
 
-          levelProducerPair.second().runForeach(output::println, materializer).
-            // close resource when the group stream is completed
-              onComplete(new OnComplete<BoxedUnit>() {
-              @Override
-              public void onComplete(Throwable failure, BoxedUnit success) throws Exception {
-                output.close();
-              }
-            }, system.dispatcher());
-        }, materializer).onComplete(new OnComplete<BoxedUnit>() {
-          @Override
-          public void onComplete(Throwable failure, BoxedUnit success) throws Exception {
-            system.shutdown();
-          }
-        }, system.dispatcher());
+    FileIO.fromFile(inputFile).
+      // parse bytestrings (chunks of data) to lines
+      via(Framing.delimiter(ByteString.fromString(System.lineSeparator()), 512, FramingTruncation.ALLOW)).
+      map(ByteString::utf8String).
+      // group them by log level
+      map((line) -> {
+        final Matcher matcher = loglevelPattern.matcher(line);
+        if (matcher.find())
+          return new Pair<>(matcher.group(1), line);
+        else
+          return new Pair<>("UNKNOWN", line);
+      }).
+      // write lines of each group to a separate file
+      runForeach((levelProducerPair) -> {
+        outputs.get(levelProducerPair.first()).println(levelProducerPair.second());
+      }, materializer).handle((ignored, failure) -> {
+        outputs.forEach((key, writer) -> {
+          writer.close();
+        });
+        system.terminate();
+         return NotUsed.getInstance();
+      });
```

Comment:
Yeah, I thought that it was more important to get this up to 2.4.6 than to wait for a new way to do the exact same, in this use case it wasn't very useful anyways.